### ISSUE TITLE
Large images look pixelated

### ISSUE THREAD

**Issue by wannaskoo** [hardcore-sushi/DroidFS#81]
OS: Android 11
Device: OnePlus 6
Platform: ARM64
DroidFS version: 1.7.2

Hi, it's me, the friendly neighbourhood big images guy.
Anyway, i've got a new problem. Every image ≥ 5MB (not just PNGs) looks pixelated when i try to view it. 
Now, i'm not a developer by any stretch of imagination, but i tried building DroidFs without this commit:
https://github.com/hardcore-sushi/DroidFS/commit/bd4c935c4c3919e119955ba120ac259ff5c77f48
And this:
https://github.com/hardcore-sushi/DroidFS/commit/23d017780fd962224ee0f211324c3e76654fc8e3
And it worked! I can see them now.
I'm sorry for not being able to help more (you know, actual code and stuff).

**Comment by hardcore-sushi** [comment]
Hum.... the commits you removed were here to fix similar bugs 😆. But it works fine for me. Can you share some screenshots and also images that cause the bug please ? (If your pictures are confidential, can you try to reproduce the bug with other large images?)

**Comment by wannaskoo** [comment]
[Test.zip](https://github.com/hardcore-sushi/DroidFS/files/7913914/Test.zip)

I tried with two other images, One is VERY heavy, the other is just barely over 5MB, which is enough to trigger the bug. I also included screenshots as you requested.

**Comment by hardcore-sushi** [comment]
That's strange, it works fine for me:
![Screenshot_DroidFS](https://user-images.githubusercontent.com/68192412/150593856-5dfe5fc9-7f4f-4a4b-9be8-47eee6877fb2.png)
But it seems someone else has the same issue: https://forge.chapril.org/hardcoresushi/DroidFS/issues/3
Does it work when you remove only bd4c935 or do you also need to remove 23d0177?
Are you OK to try a debug APK?

**Comment by wannaskoo** [comment]
Yeah, i'm ok with trying a debug apk. 
I needed to remove both commits if i remember correctly, i don't have my pc with me right now to check, sorry

**Comment by hardcore-sushi** [comment]
OK, here is a debug APK. Please install it and send the logcat you get when you reproduce the bug.
```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

Download link: https://drop.chapril.org/download/dd8fa03b4705ab10/#dmsiBqOhA0tTfm7m0NTqQA
SHA256: c019be2998a4921630064b5f987ce929eaa4f36e3e6f19450093cfcb184b0895
-----BEGIN PGP SIGNATURE-----

iQIzBAEBCgAdFiEEvVYhR557dNNqQFvoAH+EEgEHGR4FAmHsSmgACgkQAH+EEgEH
GR6Ggw//aVGRqhqnT+kbhKy7h82G12eeEZkZ6BW6mTEq1Ou/vegnKe7PhaWcUhVI
p2k8Ss9zYJ1D7/q3HoSi304+jWliI8nAMnUkc4vjj8eo2WZTGnLQVnYTBkjMhyTi
TZ+gh84WQeQEozp1cpOINO1m0AyydsWvmVsf/5ep0RGwht/Bt7FBWXDi1yatCPAC
W2t4oJ1oPbO9D2J2s57WMrjQyyTrLuUHXU84GFUkjlMENCbGAiy0ZXpTwOPaoq9r
Y6WKKLmfc1V+0TmukOjnYnhrRaHZaFyR8NccUdqEdJSMwO6j5AjHSjU1yLwXYCpa
XgpLSUrjmPhpG/AILVRq8T2uXCxfKpQ7tPH6mEdVX3P9S3Heae1xacdzE5k/Y4L6
w5DKFeTmL0INX1bAderAqSYeULzLqFzxlzQ48v2dA96nepDYpudHgn9dVUr4TS6H
IU+sLVuzafoDJTrMdKK4F2qBb+46v++eT9RDeSfvx1cyulSuCtUgzAxUwN7vcli6
tWtZ8D8/uIrQ6ftAU3LjuqYdNtlnXZQSSNyb6vlBOAxIZWf3vpWGVx0/h7t1srsK
e+6JGsOYMzHnywTt0SUQnip3UwmB9FezQC3Kh48Q7xgnW7O76PWw03CgOjjkSM2E
KQcvR+gHretPStUM0bXaTiuRqqgr5iY4TIA9oGXtcyxYmBEvieg=
=SF3u
-----END PGP SIGNATURE-----
```

**Comment by wannaskoo** [comment]
[logcat_01-23-2022_05-14-37.txt](https://github.com/hardcore-sushi/DroidFS/files/7920346/logcat_01-23-2022_05-14-37.txt)

Here's the logcat, what i did to reproduce the bug was: open app -> press "open a volume" button -> pick volume from SAF -> insert password (also with annoying prompt from my password manager lol) -> open image 1 -> open image 2 -> switch apps and stop logcat.
The strange thing is that i did not get the bug. Both test images looked un-pixelated and loaded perfectly fine.

**Comment by hardcore-sushi** [comment]
OK I found the bug. I published the patch: f546e64c345f8de9209ed500298b11194fde2bd0
The next DroidFS release should work fine.
